### PR TITLE
Updating dependencies and cleaning the rebar.lock file

### PR DESCRIPTION
**Pull request by NelsonVides**
This PR updates dependencies, and their dependencies, in an organised manner. Also reorders a bit the declaration of the dependencies, to at least put those libraries that only parse or transform data, have no side-effects, nor dependencies of their own, first (jiffy, exml, tomerl, jid, base16).

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/91501) / small_tests / 937c626657afd0fc159748c916ae87bddda2fec5
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3425/91501/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3425/91501/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-01_13.09.49/lib.mongooseim.logs/run.2021-12-01_13.09.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/91526) / small_tests / 652daf41c772230b4ef2ba38dbec9ae48b960a34
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3425/91526/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3425/91526/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-01_13.22.50/lib.mongooseim.logs/run.2021-12-01_13.22.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/91565) / small_tests / d4f46b20dc6e5938035d8c7d44c7289a39e74efa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3425/91565/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3425/91565/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-01_14.05.41/lib.mongooseim.logs/run.2021-12-01_14.05.41/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/91635) / small_tests / 3dbbdd42295dff0b357b02a8827006b4b7f38605
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3425/91635/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3425/91635/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-01_16.12.36/lib.mongooseim.logs/run.2021-12-01_16.12.37/suite.log.html)

File: rebar.config [hunk 1]
```diff
@@ -50,21 +50,32 @@
...
+  {fusco, "0.1.1"},
   {lasse, "1.2.0"},
-  {worker_pool, "4.0.3"},
+
+  {fast_scram, "0.4.3"},
+  {fast_tls, "1.1.13"},
+  {cache_tab, "1.0.29"},
+
+  {segmented_cache, "0.1.1"},
+  {idna, "6.1.1"},
```
Comment (gustawlippa) [inline rebar.config:76]: This could be maybe moved to the top?

File: rebar.config [hunk 2]
```diff
@@ -75,37 +86,28 @@
   {eodbc, {git, "https://github.com/arcusfelis/eodbc.git", {branch, "master"}}},
   {tirerl, "1.1.0"},
 
-  {cache_tab, "1.0.28"},
-  {jid, "1.0.0", {pkg, mongoose_jid}},
-
   {proper, "1.4.0"},
   {meck, "0.8.13"},
```
Comment (gustawlippa) [inline rebar.config:90]: Not this PR, but how is it that we need these two... 🤔

Comment (NelsonVides) [reply rebar.config:90]: Those are funny ones. They could be declared as dependencies in the test profile for rebar, but, big tests are actually rpc-ing to them to do some black magic 🤷🏽‍♂️

File: rebar.config (hunk 1 above)
Comment (NelsonVides) [reply rebar.config:76]: idna maybe, it's a pure library with no state and only one irrelevant dependency, but segmented_cache has a state (persistent term and a bunch of ets tables), I'd rather keep it like, next to cache_tab, another caching library 🤔

Comment (gustawlippa) [reply rebar.config:76]: I was thinking about idna, they even removed the dependency in the repository, but didn't publish a new package without it.

Comment (NelsonVides) [reply rebar.config:76]: Oh, so we gotta ping benoit and ask 🤔

File: rebar.config (hunk 2 above)
Comment (arcusfelis) [reply rebar.config:90]: yea, but they don't go into the release build, so it's fine

**Review by arcusfelis — APPROVED** [review]
good to me, we cannot wait for the external devs to update stuff on hex :)
Once they are ready, we will make a new PR.

**Pull request merged**